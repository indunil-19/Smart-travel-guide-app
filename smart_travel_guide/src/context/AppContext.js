import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}