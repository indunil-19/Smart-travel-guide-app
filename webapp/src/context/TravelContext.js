import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 