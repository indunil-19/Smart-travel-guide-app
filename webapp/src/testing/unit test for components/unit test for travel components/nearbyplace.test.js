import React from "react"
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter,Router } from "react-router-dom";
import { TravelProvider } from "../../../context/TravelContext";
import "@testing-library/jest-dom/extend-expect";
import 'whatwg-fetch'
import NearByPlaceCard from "../../../components/TravelPlanApp/nearbyplace";
import { getByTestId, waitFor } from "@testing-library/dom";

describe('nearby place component test', () => {
    
    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><NearByPlaceCard/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><NearByPlaceCard  photo={["123"]} index={"1"} name={"example hotel"} address="example address"  types={[]} rating="5" /></TravelProvider></MemoryRouter>,div)
       waitFor(()=>{
           expect(getByTestId("card")).toHaveTextContent("example hotel")
       })
       
    })
})
