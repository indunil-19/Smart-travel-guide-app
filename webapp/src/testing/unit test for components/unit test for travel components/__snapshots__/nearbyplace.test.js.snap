// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nsvbar component test should match snapshot 1`] = `
<div
  className="css-bp8v6l"
  data-testid="card"
>
  <div
    className="css-pp0aaf"
  >
    <img
      className="chakra-image__placeholder css-joy71d"
    />
    <div
      className="chakra-stack css-t1lozp"
    >
      <span
        className="chakra-avatar css-1ruhx4w"
      >
        <div
          aria-label="NaN"
          className="chakra-avatar__initials css-0"
          role="img"
        >
          N
        </div>
      </span>
      <h4
        className="css-t45tyy"
      >
         
      </h4>
      <div
        className="css-bgvdb6"
      />
      <div
        className="chakra-stack css-84zodg"
      >
            
      </div>
      <div
        className="chakra-stack css-84zodg"
      >
        <svg
          className="chakra-icon css-4i8mdq"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
            fill="currentColor"
          />
        </svg>
        <svg
          className="chakra-icon css-4i8mdq"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
            fill="currentColor"
          />
        </svg>
        <svg
          className="chakra-icon css-4i8mdq"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
            fill="currentColor"
          />
        </svg>
        <svg
          className="chakra-icon css-4i8mdq"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
            fill="currentColor"
          />
        </svg>
        <svg
          className="chakra-icon css-4i8mdq"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
            fill="currentColor"
          />
        </svg>
        <div
          className="css-0"
        >
          rating 
        </div>
      </div>
    </div>
  </div>
</div>
`;
